src/NJsonSchema/Generation/JsonSchemaGenerator.cs

```diff
@@ -531,6 +539,27 @@ public void ApplyPropertyAnnotations(JsonSchema4 jsonProperty, Type parentType,
                     jsonProperty.MaxLength = stringLengthAttribute.MaximumLength;
                 }
             }
+
+            dynamic dataTypeAttribute = TryGetAttribute(attributes,
+                "System.ComponentModel.DataAnnotations.DataTypeAttribute");
+            if (dataTypeAttribute != null)
+            {
+                var knownFormats = new Dictionary<string, string>

```

Comment:
You should make this static in the class so that it is not created multiple times...